radis/lbl/calc.py

```diff
@@ -28,17 +28,321 @@
+​
+    Refer to the online :ref:`Examples <label_examples>` for more cases, and to 
+    the :ref:`Spectrum page <label_spectrum>` for details on post-processing methods. 
+​
+    See Also
+    --------
+    
+    :class:`~radis.lbl.factory.SpectrumFactory`, 
+    the :ref:`Spectrum page <label_spectrum>`
+    '''
+    
+    if not isinstance(molecule,dict):
+        s = _calc_spectrum(wavenum_min=wavenum_min,
+                           wavenum_max=wavenum_max,
+                           wavelength_min=wavelength_min,
+                           wavelength_max=wavelength_max,
+                           Tgas=Tgas,
+                           Tvib=Tvib,
+                           Trot=Trot,
+                           pressure=pressure,
+                           overpopulation=overpopulation,
+                           molecule=molecule,
+                           isotope=isotope,
+                           mole_fraction=mole_fraction,
+                           path_length=path_length,
+                           databank=databank,
+                           medium=medium,
+                           wstep=wstep,
+                           broadening_max_width=broadening_max_width, 
+                           lineshape_optimization=lineshape_optimization,
+                           name=name,
+                           use_cached=use_cached,
+                           verbose=verbose,
+                           **kwargs)
+        return s
+    else: # The "molecules" dictionary is read, instead of the possible other inputs ("molecule", "mole_fraction", ...)    
+        # Test that all overwritten parameters are left as default:
+        try:
+            assert mole_fraction == 1
+            assert overpopulation is None 
+            assert isotope is None
+        except AssertionError:
+            raise ValueError('mole_fraction, overpopulation and isotope '+\
+                             'should be left as default if using a dictionary of '+\
+                                 'molecules')
+        
+        s = []
+        for mol in molecule.keys():
+            s.append(_calc_spectrum(wavenum_min=wavenum_min,
+                                   wavenum_max=wavenum_max,
+                                   wavelength_min=wavelength_min,
+                                   wavelength_max=wavelength_max,
+                                   Tgas=Tgas,
+                                   Tvib=Tvib,
+                                   Trot=Trot,
+                                   pressure=pressure,
+                                   overpopulation=molecule[mol]['overpopulation'],
+                                   molecule=mol,
+                                   isotope=molecule[mol]['isotopes'],
+                                   mole_fraction=molecule[mol]['mole_fraction'],
+                                   path_length=path_length,
+                                   databank=molecule[mol]['databank'],
```

Comment:
The default value should be fetch if databank field is not mentioned. I would prefer `molecule[mol].get('databank', "fetch")` for all the dict reference.

Reply:
Sounds good. Could be `molecule[mol].get('databank', databank)` to avoid hardcoding the default.